docs/txp.md

```diff
@@ -0,0 +1,334 @@
+   because data is unknown. This behaviour depends on which type of
+   data we are processing, each particular case is described in more
+   details below.
+
+The behaviour depends on two protocol versions: version used by this
+software and last adopted version. We verify that data in blocks is
+known if protocol version used by this software is greater than or
+equal to the adopted version. That's because in this case:
+
+1. Authors of this software are aware of the adopted version.
+2. Each issued block must be formed with respect to adopted version.
+
+Comparison is quite tricky here. Table below demonstrates it.
+
+| Our   | Adopted | Check? |
+| ----- | ------- | ------ |
+| 1.2.3 |  1.2.3  | Yes    |
+| 1.2.3 |  1.2.4  | No     |
+| 1.2.3 |  1.2.2  | No     |
+| 1.2.3 |  1.3.2  | No     |
+| 1.2.3 |  1.1.1  | Yes    |
+| 2.2.8 |  1.9.9  | Yes    |
+
+If `(major, minor)` of our version is greater than of adopted
+one, then check is certainly done. If it's equal, then check is
+done only if `alt` component is the same as adopted one. In
+other cases (i. e. when our `(major, minor)` is less than from
+adopted version) check is not done.
+
+Let's assume that we have already compared these two versions and know
+whether we permit unknown data in blocks. Let's call a flag which
+determines whether all data must be known `verifyAllIsKnown`.
+
+Note: when we say that attributes must be known, it means that
+unparsed fields must be empty.
+
+### Outputs checks
+
+For each output address we perform the following checks:
+1. If `verifyAllIsKnown` is `True`, address' attributes must be known.
+2. If `verifyAllIsKnown` is `True`, address' type must be known.
+3. Address can't be a redeem address.
+
+### Consistency check
+
+There is only one consistency check: number of inputs in a transaction
+must be the same as number of witnesses for this transaction.
+
+### Transaction attributes check
+
+If `verifyAllIsKnown` is `True`, transaction's attributes must be known.
+
+### Inputs checks
+
+Other checks heavily depend on whether there is at least one input
+with unknown type. There are three cases:
+
+1. There is such input and `verifyAllIsKnown` is `True`. In this case
+   transaction is immediately considered invalid.
+2. There is such input and `verifyAllIsKnown` is `False`. It means
+   that there was a protocol update not known to our software. In this
+   case we can't know whether that input is valid and how much value
+   it has. So we perform only basic
+   checks: [outputs checks](#outputs-checks)
+   and [consistency check](#consistency-check). Note that we don't
+   perform
+   [transaction's attributes check](#transaction-attributes-check),
+   because `verifyAllIsKnown` is `False`.
+3. All inputs have known format (i. e. `TxInUtxo`). In this case
+   (which should be the most common one) we can check all inputs'
+   legitimacy, compare input and output sums, check fee, etc. All
```

Comment:
> inputs' legitimacy, compare input and output sums, check fee
>
> etc.

Need more detailed explanation. Some of these checks are described later, but not all. E.g. what's legitimacy. And there shouldn't be "etc.". I would propose to have here bullet-list and in subsections of this section show every check.

Reply:
> All inputs must exist in UTXO

Important to note that this is simple hashtable check w/o modification of the hashtable

Reply:
I wrote that these checks are described below and rephrased some stuff here.